### PR TITLE
#1792 Add comments

### LINKED ISSUE #1792: Doxygen comments
update to ubuntu 20 shows some missing doxygen comments, previously undetected

### PR DESCRIPTION
**Pull request by lecourtoisflo**
closes #1792

### CODE REVIEW

File: util/envDynawo.sh [hunk 1]
```diff
@@ -1194,13 +1194,16 @@ build_doxygen_doc() {
 
 test_doxygen_doc() {
   if [ -f "$DYNAWO_INSTALL_DIR/doxygen/warnings.txt"  ] ; then
-    nb_warnings=$(wc -l $DYNAWO_INSTALL_DIR/doxygen/warnings.txt | awk '{print $1}')
+    rm -f $DYNAWO_INSTALL_DIR/doxygen/warnings_filtered.txt
+    # need to filter "return type of member (*) is not documented" as it is a doxygen bug detected on 1.8.17 that will be solbed in 1.8.18
```
Comment (gautierbureau) [inline util/envDynawo.sh:1198]: solbed -> solved

**Review by gautierbureau — APPROVED** [review]

File: util/envDynawo.sh (hunk 1 above)
Comment (lecourtoisflo) [reply util/envDynawo.sh:1198]: OK

File: util/envDynawo.sh [hunk 2]
```diff
@@ -1194,13 +1194,16 @@ build_doxygen_doc() {
 
 test_doxygen_doc() {
   if [ -f "$DYNAWO_INSTALL_DIR/doxygen/warnings.txt"  ] ; then
-    nb_warnings=$(wc -l $DYNAWO_INSTALL_DIR/doxygen/warnings.txt | awk '{print $1}')
+    rm -f $DYNAWO_INSTALL_DIR/doxygen/warnings_filtered.txt
+    # need to filter "return type of member (*) is not documented" as it is a doxygen bug detected on 1.8.17 that will be solved in 1.8.18
+    cat $DYNAWO_INSTALL_DIR/doxygen/warnings.txt | grep -v -e "return type of member .* is not documented" > $DYNAWO_INSTALL_DIR/doxygen/warnings_filtered.txt
```
Comment (rosiereflo) [inline util/envDynawo.sh:1199]: Shouldn't we make sure that we filter only the static methods? (not sure there is a way to do so though)

**Review by rosiereflo — CHANGES_REQUESTED** [review]

File: util/envDynawo.sh (hunk 2 above)
Comment (lecourtoisflo) [reply util/envDynawo.sh:1199]: The line in the doxygen warnings doesn't contain the attribute static or virtual so we cannot

Comment (rosiereflo) [reply util/envDynawo.sh:1199]: It means that all return types issue will be filtered, even the real ones... Maybe with a file that contains all the files to filter?

**Review by rosiereflo — APPROVED** [review]

**Pull request merged**